### DISCUSSION TITLE
error: Failed to load the submitted data due to invalid formatting

### DISCUSSION THREAD

**Discussion by dev-ton** [pocketbase/pocketbase#1935] [Q&A] [answered]
Hello, I'm getting this error when I try to upload file. I am sending the type File object. See below.

my code:
```
const addToPb = async () => {
  const file = fields.value[0].fileInput
  if (file) {
    formData.append('files', file)
  }
  formData.append('title', fields.value[0].textInput)
  try {
    const record = await pb.collection('animations').create(formData)
    return record
  } catch (error) {
    return { error: error.message }
  }
}
```

the file payload:
```
FileList {0: File, length: 1}
0: File
{
lastModified: 1676471349792
lastModifiedDate: Wed Feb 15 2023 15:29:09 GMT+0100 (Central European Standard Time) {}
name: "file.pdf"
size: 13106816
type: "application/pdf"
webkitRelativePath: ""
}
```

pls help 🙏

**Comment by ganigeorgiev** [comment, marked as answer]
`file` must be a `File` or `Blob` instance and if I'm reading correctly the code sample you are passing the DOM input element. You need to append the `File` entries of `fields.value[0].fileInput.files`. For a plain js example you can check https://pocketbase.io/docs/files-handling/#uploading-files.

**Reply by dev-ton** [reply]
Yeah, I was feeding it a full Fileset. Now it works perfectly. Thanks.
